tls/s2n_connection.c

```diff
@@ -534,10 +534,20 @@ void *s2n_connection_get_ctx(struct s2n_connection *conn)
 
 int s2n_connection_release_buffers(struct s2n_connection *conn)
 {
-    GUARD(s2n_stuffer_release_if_empty(&conn->out));
-    GUARD(s2n_stuffer_release_if_empty(&conn->in));
+    notnull_check(conn);
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(&conn->out));
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(&conn->in));
 
-    return 0;
+    if(conn->out.blob.data != NULL && s2n_stuffer_is_consumed(&conn->out)) {
+        GUARD(s2n_stuffer_resize(&conn->out, 0));
+    }
+    if(conn->in.blob.data != NULL && s2n_stuffer_is_consumed(&conn->in)) {
```

Comment:
why the notnull check?

Reply:
Or why not move the check into the `is_consumed`?

Reply:
I don't want to raise an error if it's null, I just don't need to resize to 0 if the data is already null. @danielsn

Reply:
That is already effectively a no-op. https://github.com/awslabs/s2n/blob/master/stuffer/s2n_stuffer.c#L89

Having a notnull check here is premature optimization, and hides the actual logic of the code.

Reply:
I removed the nullness checks and replaced with the appropriate checks.